azurerm/internal/services/securitycenter/server_vulnerability_assessment_resource.go

```diff
@@ -0,0 +1,167 @@
+package securitycenter
+
+import (
+	"fmt"
+	"log"
+	"time"
+
+	"github.com/Azure/azure-sdk-for-go/services/preview/security/mgmt/v3.0/security"
+
+	"github.com/hashicorp/terraform-plugin-sdk/helper/schema"
+	"github.com/terraform-providers/terraform-provider-azurerm/azurerm/helpers/azure"
+	"github.com/terraform-providers/terraform-provider-azurerm/azurerm/helpers/tf"
+	"github.com/terraform-providers/terraform-provider-azurerm/azurerm/internal/clients"
+	"github.com/terraform-providers/terraform-provider-azurerm/azurerm/internal/timeouts"
+	"github.com/terraform-providers/terraform-provider-azurerm/azurerm/utils"
+)
+
+func resourceServerVulnerabilityAssessment() *schema.Resource {
+	return &schema.Resource{
+		Create: resourceServerVulnerabilityAssessmentCreate,
+		Read:   resourceServerVulnerabilityAssessmentRead,
+		Delete: resourceServerVulnerabilityAssessmentDelete,
+
+		Importer: &schema.ResourceImporter{
+			State: schema.ImportStatePassthrough,
+		},
+
+		Timeouts: &schema.ResourceTimeout{
+			Create: schema.DefaultTimeout(5 * time.Minute),
+			Read:   schema.DefaultTimeout(5 * time.Minute),
+			Update: schema.DefaultTimeout(5 * time.Minute),
+			Delete: schema.DefaultTimeout(5 * time.Minute),
+		},
+
+		Schema: map[string]*schema.Schema{
+			"virtual_machine_id": {
+				Type:         schema.TypeString,
+				Required:     true,
+				ForceNew:     true,
```

Comment:
Seems ServerVulnerabilityAssessment can be updated from server A to server B. So ForceNew isn't required. Could you double check whether it could be updated in-place?

Reply:
I will take a look at this.

Reply:
@neil-yechenwei I can not see how I would do a in-place update with the API.The API has the method **CreateOrUpdate**, but that method takes the parameters resourceGroupName, resourceNamespace, resourceType, and resourceName to identify the **server** to create or update the vulnerability assessment on. So I do not think it is possible to specify a specific vulnerability assessment and move that. Instead you need to delete the old and install a new.
Do you agree, or am I missing something?

Reply:
Yes. I assume you're right.